### ISSUE TITLE
Add P2 to Detect

### ISSUE THREAD

**Issue by Spratumn** [ultralytics/ultralytics#17823]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I added an input in the detect, and during training, the GPU memory usage keeps increasing. Moreover, after a certain period of training, one particular epoch becomes extremely slow.
```
- [[2, 16, 19, 22], 1, Detect, [nc]] # Detect(P2, P3, P4, P5)
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Spratumn, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, we kindly ask that you provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it. Please share all relevant code, system details, and clear instructions to reproduce the issue. This will allow us to investigate the GPU memory increase and slow epoch behavior more efficiently.

If this is a custom training ❓ Question, please provide as much information as possible, such as model modifications, dataset image examples, training logs, and any relevant configurations. Ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://ultralytics.com/discord) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/ultralytics) to share knowledge with the community.

## Upgrade

We recommend upgrading to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify the issue is not resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in a variety of up-to-date verified environments (preinstalled with all dependencies like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/)):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help guide you in the right direction 😊. An Ultralytics engineer will also review your issue and provide assistance soon.

**Comment by Y-T-G** [comment]
This isn't P2. It's the second layer of the network which is almost like passing the original image because it hardly went through any convolution to be downsampled. It will consume a lot of memory.

**Comment by Y-T-G** [comment]
You can see how P2 output is created here.

https://github.com/ultralytics/ultralytics/blob/7c97ed19372655a8b3d9050fc0f2ffb2e6a31276/ultralytics/cfg/models/v8/yolov8-p2.yaml#L4

**Comment by Spratumn** [comment]
Thank for your replay, I used the model configuration file [ultralytics/ultralytics/cfg/models/v8/yolov8-p2.yaml](ultralytics/ultralytics/cfg/models/v8/yolov8-p2.yaml) but the same issue occurs. What I want to express is that when providing 4 inputs to Detect, there is a problem of continuous GPU memory increase during model training. I verified this issue with a very small batch size (2). At the beginning of training, the first epoch takes about 200 iterations with less than 2G of GPU memory usage. By around 2000 iterations, the GPU memory usage increases to 4.7G, and it continues to grow until the entire GPU memory is occupied, after which the training speed significantly decreases.

**Comment by Spratumn** [comment]
@Y-T-G  Thank for your replay, I used the model configuration file [ultralytics/ultralytics/cfg/models/v8/yolov8-p2.yaml] (https://github.com/ultralytics/ultralytics/issues/ultralytics/ultralytics/cfg/models/v8/yolov8-p2.yaml) but the same issue occurs. What I want to express is that when providing 4 inputs to Detect, there is a problem of continuous GPU memory increase during model training. I verified this issue with a very small batch size (2). At the beginning of training, the first epoch takes about 200 iterations with less than 2G of GPU memory usage. By around 2000 iterations, the GPU memory usage increases to 4.7G, and it continues to grow until the entire GPU memory is occupied, after which the training speed significantly decreases.

**Comment by Y-T-G** [comment]
What's your training command?

**Comment by Spratumn** [comment]
@Y-T-G 
```
from ultralytics import YOLO

if __name__ == '__main__':
    # 加载模型
    model = YOLO("yolov8-p2.yaml")  # 从头开始构建新模型
    model.train(data="mydata.yaml")  # 训练模型
```
